File: Examples/Calculator/Calculator/CalcErrorDelegate.m
//
//  CalcErrorDelegate.m
//  Calculator
//
//  Created by William Woody on 4/1/19.
//  Copyright © 2019 Glenview Software. All rights reserved.
//

#import "CalcErrorDelegate.h"

@implementation CalcErrorDelegate

- (void)errorFrom:(CalcParser *)yacc line:(NSInteger)line
		column:(NSInteger)column filename:(NSString *)fname
		errorCode:(NSInteger)errorCode
		data:(NSDictionary<NSString *, id<NSObject>> *)data
{
	NSString *err;

	switch (errorCode) {
		case ERROR_SYNTAX:
			err = @"Syntax Error";
			break;
		case ERROR_MISSINGTOKEN:
			err = [NSString stringWithFormat:@"Missing token (%@)",data[@"token"]];
			break;
		case ERROR_MISSINGTOKENS:
			err = [NSString stringWithFormat:@"Missing tokens (%@)",[data[@"tokens"] description]];
			break;
		default:
			err = @"Unknown error";
			break;
	}

	NSLog(@"%@ at %d:%d",err,(int)line,(int)column);
}

@end

File: Examples/CalcErrorSwift/CalcErrorSwift/CalcErrorDelegate.swift
//
//  CalcErrorDelegate.swift
//  CalcErrorSwift
//
//  Created by William Woody on 4/22/19.
//  Copyright © 2019 Glenview Software. All rights reserved.
//

import Cocoa

class CalcErrorDelegate: CalcParserError {
	func error(parser: CalcParser, line: Int, column: Int, filename: String?, errorCode: Int, data: [String : AnyObject]?) {

		// Report error
		print("Error: ", errorCode, " at ", line, ":", column)
	}
}

